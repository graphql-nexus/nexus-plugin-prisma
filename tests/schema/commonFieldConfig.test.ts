import * as Nexus from 'nexus'
import { generateSchemaAndTypes } from '../__utils'

it('forwards description & deprecation on all field types', async () => {
  const datamodel = `
    model User {
      id  Int @id @default(autoincrement())

      name String
      emailConfirmed Boolean
      birthYear Int
      heightM Float

      favouriteColor  Color

      latestPostId Int?
      latestPost Post?  @relation("latestPost", fields: [latestPostId], references: [id])
      posts Post[] @relation("posts")
    }

    enum Color {
      Red
      Green
      Blue
    }

    model Post {
      id  Int @id @default(autoincrement())
      title String
      body String
      userId Int
      user User @relation("posts", fields: [userId], references: [id])
      latestPostUsers User[] @relation("latestPost")
    }
  `

  const Post = Nexus.objectType({
    name: 'Post',
    definition(t: any) {
      t.model.id()
    },
  })

  const User = Nexus.objectType({
    name: 'User',
    definition(t: any) {
      t.model.id({ description: 'The unique id', deprecation: 'never' })
      t.model.name({ description: "The user's full name", deprecation: 'who needs names' })
      t.model.heightM({ description: "The user's height, in meters", deprecation: "size doesn't matter" })
      t.model.favouriteColor({ description: "The user's favorite color", deprecation: 'no longer required' })
      t.model.posts({ description: "All the user's blog posts", deprecation: 'no more blog' })
      t.model.latestPost({ description: "The user's last blog post", deprecation: 'not here anymore' })
    },
  })

  const { schemaString: schema } = await generateSchemaAndTypes(datamodel, [User, Post])

  expect(schema).toMatchSnapshot('commonFieldsOutput')
})
