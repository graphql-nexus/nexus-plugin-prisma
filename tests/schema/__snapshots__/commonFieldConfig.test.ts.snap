// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forwards description & deprecation on all field types: commonFieldsOutput 1`] = `
"type User {
  """The unique id"""
  id: Int! @deprecated(reason: "never")

  """The user's full name"""
  name: String! @deprecated(reason: "who needs names")

  """The user's height, in meters"""
  heightM: Float! @deprecated(reason: "size doesn't matter")

  """The user's favorite color"""
  favouriteColor: Color! @deprecated(reason: "no longer required")

  """All the user's blog posts"""
  posts(first: Int, last: Int, before: PostWhereUniqueInput, after: PostWhereUniqueInput): [Post!]! @deprecated(reason: "no more blog")

  """The user's last blog post"""
  latestPost: Post @deprecated(reason: "not here anymore")
}

type Post {
  id: Int!
}

enum Color {
  Red
  Green
  Blue
}

input PostWhereUniqueInput {
  id: Int
}

type Query {
  ok: Boolean!
}"
`;
